lib/gateway/actionParams.js

```diff
@@ -0,0 +1,30 @@
+const log = require('../logger').policy;
+class ActionParams {
+}
+
+// TODO: this may be not the best way,
+// but it seems to be the only one to allow reuse this code in plugins without requiring SDK
+ActionParams.prototype.getCommonAuthCallback = function (req, res, next) {
+  return (err, user, info) => {
+    // passThrough allows auth to fail
+    // (hoping that next auth policy will be able to do auth)
+    if (!user && (this.passThrough || this.passthrough)) {
+      log.debug('auth passThrough enabled, continuing pipeline', err, info);
+      return next();
+    }
+
+    if (err) {
+      return next(err);
+    }
+    if (user) {
+      req.logIn(user, (err) => next(err));
+    } else if (info && info.unauthorized) {   // user is not authorized
```

Comment:
I might be wrong here but the way I read this part of code is to return `403` if `info.unauthorized` is defined. Shouldn't it be the opposite?

Reply:
if `info.unauthorized` IS *true* return 403. what is wrong here ?